File: src/2144.minimum-cost-of-buying-candies-with-discount.2248/solution.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minimumCost = function (cost) {
    cost.sort((a, b) => b - a);
    let result = 0;
    let index = 0;
    while (index + 1 < cost.length) {
        result += cost[index++];
        result += cost[index];
        index += 2;
    }
    while (index < cost.length) {
        result += cost[index++];
    }
    return result;
};

File: src/2144.minimum-cost-of-buying-candies-with-discount.2248/solution.java
class Solution {
    public int minimumCost(int[] cost) {
        Arrays.sort(cost);
        int result = 0;
        int index = cost.length-1;
        while(index>-1){
            result += cost[index--];
            if(index>-1){
                result += cost[index--];
            }
            index--;
        }
        return result;
    }
}